component/src/main/java/org/wso2/extension/siddhi/io/file/FileSink.java

```diff
@@ -128,6 +139,11 @@ protected void init(StreamDefinition streamDefinition, OptionHolder optionHolder
         if (Constants.TRUE.equalsIgnoreCase(append)) {
             properties.put(Constants.APPEND, append);
         }
+        String mapType = streamDefinition.getAnnotations().get(0).getAnnotations().get(0).getElements().get(0)
+                .getValue();
+        addEventSeparator = !mapType.equalsIgnoreCase("csv") &&
+                Boolean.parseBoolean(optionHolder.validateAndGetStaticValue(Constants.ADD_EVENT_SEPARATOR,
```

Comment:
Check if ADD_EVENT_SEPARATOR exist? 
If it exists use that value. 
only if it does not exist use the defaults (true / in csv case use false)

Reply:
Fixed in 2eb346c